### DISCUSSION TITLE
[Vulnerability] Is AdGuard Home prone to log4shell?

### DISCUSSION THREAD

**Discussion by privacy-advo** [AdguardTeam/AdGuardHome#3939] [Q&A] [answered]
Hi, 
are some AdGuard products, especially AdGuard Home, prone to CVE-2021-44228 aka log4shell?
Thanks

**Comment by sachinwadhwa** [comment]
It's "Log4j" which is Java based utility. Adguard is based on Go language and I don't think log4j is used.

**Comment by privacy-advo** [comment]
Yes, for sure. But it's always nice to have a feedback from the developers concerning their wide array of products. // And the vulnerability is called _log4shell_.

**Comment by ainar-g** [comment, marked as answer]
As @sachinwadhwa said, we use Go, and so don't use the `log4j` library.  Our plain text logging is very simple, and we don't use LDAP, JNDI, or any similar technologies in our logs.

**Reply by privacy-advo** [reply]
Thanks @ainar-g
